CHANGELOG.rst

```diff
@@ -4,6 +4,12 @@ Changelog
 in development
 --------------
 
+0.9.0
+-----
+* Update dependencies (change)
+* Upgrade from hubot 2 to 3 (change)
+* Drop support for Node.js < 6 (change)
+* Add support for Node.js >= 6 (change)
```

Comment:
Isn't `>= 6` too much greedy way of showing it?
ATM we're pinning support to maximum node `10`. When new nodejs version will be released (say `11`), it might be confusing.

I'd propose to be more clear and list explicitly which versions are deprecated and which new versions were added.

Reply:
If I pin it to a maximum of node 10, and node 11 _doesn't_ break anything, then we will need to revisit this to update the accepted node.js version. Only specifying a minimum version relieves us of a bit of the maintenance burden.

And considering how well we update our node projects, I'm in favor of reducing our maintenance burden.

Reply:
I'll be more detailed about what dependencies were upgraded though. :)

Reply:
> If I pin it to a maximum of node 10, and node 11 doesn't break anything, then we will need to revisit this to update the accepted node.js version

By supporting specific versions which we already doing, I mean this: 
https://github.com/StackStorm/hubot-stackstorm/blob/6d8721a596eaaeaa832cccb26099a3a174d1aff4/.travis.yml#L1-L9

If we'll support node 11, - that will come with another `hubot-stackstorm` version update and a new Changelog entry and code change. The support statement shouldn't be that blurred.

Reply:
It's entirely acceptable (to me) to have differences between "works with", "supported", and "tested" though.

* We can test for more versions than we officially support (I've done this on some projects), because developers who run older versions of dependencies know that they are behind the curve.
* The code may work with more versions than we officially support or test - older or newer versions of node fall into this category.

So yes, once we officially support node 11, we should test with that, add it to the changelog, and officially support it, but if things might still work with node 11, I don't see any reason to explicitly limit the code compatibility just to node 10.

This might be a good time to put the `node` version (which tests the most recent version of node.js at build time) into the `allow_failures` part of the Travis config. That may be a good middle ground?

Reply:
Yes, code compatibility is good :+1: 
The point is just to be very clear in Changelog/documentation what we support and what's not.

It might be also useful as you'll work on st2chatops: we pin max supported `nodejs` version in both `st2chatops` deb/rpm packages (https://github.com/StackStorm/st2chatops/blob/492b6daaafe4aa7faf83be634e38b54b744ca7d2/debian/control#L14) and `curl|bash` installer: https://github.com/StackStorm/st2-packages/blob/0f2770be1fac382267f9c9cdab27a3b1865ab9f7/scripts/st2bootstrap-deb.sh#L656-L657

Reply:
I've been wondering if we should relax our nodejs version dependency in those packages. E.g. support 6/8/10, rather than just 6.

Having a wider range might make things a little easier for folks migrating between versions. It's a pain changing versions due to need to repo changes. Makes upgrades a bit more involved.

Reply:
+1, setting supported interval of versions would work.
Just hope there won't be any compatibility issues between min (already installed for some users) and max (new installs) version as we update `st2chatops` + `hubot-stackstorm`.

Probably needs another cycle of manual testing?